aws/client.go

```diff
@@ -64,7 +64,7 @@ func NewClient(cfg Config, metadata Metadata) *Client {
 	retryer := cfg.Retryer
 	if retryer == nil {
 		// TODO need better way of specifing default num retries
-		retryer = DefaultRetryer{NumMaxRetries: 3}
+		retryer = DefaultRetryer{3, 0}
```

Comment:
This should use named struct field initialization. e.g.

```
retryer = DefaultRetryer{
    NumMaxRetries: 3,
}
```

also don't need to initialize the `MinTime` member since it `0` is its default [zero value](https://dave.cheney.net/2013/01/19/what-is-the-zero-value-and-why-is-it-useful).